TACC/Core-Portal
# task/TV3-109 Concatenate homeDir in DataFiles Breadcrumbs

## Overview

The goal for this task was to concatenate the path that displays on the `Data Files` tab for a system if the path was the homeDir of that system. The intended behavior should be the same as the current v2 portals. Since the way these paths are handled in tapis v3 is different than v2 some tweaks had to be made to ensure same functionality as v2.


## Related

* [TV3-109](https://jira.tacc.utexas.edu/browse/TV3-109)

## Changes

- Added a selector to get the systems displayed to the user to retrieve the homeDir for that system
- Added a comparison based on the homeDir to determine if the path should be added to BreadcrumbLink or not


## Testing

1. Log on to the portal and go to Data Files 
2. Confirm that only the main system name is shown on the top left initially.
3. Navigate into folders to ensure paths get appended properly
4. Click on folder names on top left to confirm navigation works properly

## UI

Before: 

<img width="1792" alt="Screen Shot 2023-02-21 at 10 15 52 AM" src="https://user-images.githubusercontent.com/23081932/220452450-8a712fe3-b91c-4a0a-9f97-4a2ae2010cd8.png">

After:

<img width="1792" alt="Screen Shot 2023-02-21 at 10 16 14 AM" src="https://user-images.githubusercontent.com/23081932/220452516-4231a94f-71aa-43ac-bed7-2a1684def7b9.png">

<img width="1792" alt="Screen Shot 2023-02-21 at 10 16 33 AM" src="https://user-images.githubusercontent.com/23081932/220452548-7eeb500a-4f08-48c1-90ce-dd9987376d26.png">





## Notes

There is a bug I found while testing this feature. For the systems Community Data and Public Data the concatenation does not work properly. I realized this is a larger issue because the correct system name is not being returned (notice in top left it is a different name) for both Community Data and Public Data. This fix might be out of scope for this task so I created another task [TV3-120 Fix incorrect system name being displayed for Community Data and Public Data](https://jira.tacc.utexas.edu/browse/TV3-120). This issue is addressed in https://github.com/TACC/Core-Portal/pull/767
